cirq-core/cirq/transformers/synchronize_terminal_measurements.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Transformer pass to move terminal measurements to the end of circuit."""
```

Comment:
Is this defined as the last moment in the circuit ?

Reply:
Yes, but the precise moment to which it get's added depends upon the value of `after_other_operations` flag.